src/nni_manager/training_service/remote_machine/sshClientUtility.ts

```diff
@@ -58,38 +58,56 @@ export namespace SSHClientUtility {
      * @param command the command to execute remotely
      * @param client SSH Client
      */
-    export function remoteExeCommand(command: string, client: Client): Promise<RemoteCommandResult> {
+    export function remoteExeCommand(command: string, client: Client, useShell: boolean = false): Promise<RemoteCommandResult> {
         const log: Logger = getLogger();
         log.debug(`remoteExeCommand: command: [${command}]`);
         const deferred: Deferred<RemoteCommandResult> = new Deferred<RemoteCommandResult>();
         let stdout: string = '';
         let stderr: string = '';
         let exitCode: number;
+        let captureStdOut = false;
 
-        client.exec(command, (err: Error, channel: ClientChannel) => {
+        const callback = (err: Error, channel: ClientChannel): void => {
             if (err !== undefined && err !== null) {
                 log.error(`remoteExeCommand: ${err.message}`);
                 deferred.reject(err);
-
                 return;
             }
 
-            channel.on('data', (data: any, dataStderr: any) => {
-                if (dataStderr !== undefined && dataStderr !== null) {
-                    stderr += data.toString();
-                } else {
-                    stdout += data.toString();
+            channel.on('data', (data: any) => {
+                if (captureStdOut) {
+                    stdout += data;
                 }
-            })
-              .on('exit', (code: any, signal: any) => {
+            });
+            channel.on('exit', (code: any) => {
                 exitCode = <number>code;
+                log.debug(`remoteExeCommand exit(${exitCode})\nstdout: ${stdout}\nstderr: ${stderr}`);
                 deferred.resolve({
-                    stdout : stdout,
-                    stderr : stderr,
-                    exitCode : exitCode
+                    stdout: stdout,
+                    stderr: stderr,
+                    exitCode: exitCode
                 });
             });
-        });
+            channel.stderr.on('data', function (data) {
+                stderr += data;
+            });
+
+            if (useShell) {
+                captureStdOut = true;
+                channel.stdin.write(`${command}\n`);
+                captureStdOut = false;
+                channel.end("exit\n");
+            }
+
+            return;
+        };
+
+        if (useShell) {
+            client.shell(callback);
```

Comment:
how to exec command in `client.shell`?

Reply:
It supports in ssh2 package, and the command will be send in 'ready' event as above lines

```js
if (useShell) {
  channel.stdin.write(`${command}\n`);
  channel.end("exit\n");
}
```